nwb_conversion_tools/datainterfaces/axonadatainterface.py

```diff
@@ -0,0 +1,316 @@
+"""Authors: Steffen Buergers"""
+import os
+import re
+import datetime
+import contextlib
+import mmap
+import numpy as np
+
+import spikeextractors as se
+
+from pynwb import NWBFile
+from pynwb.behavior import Position, SpatialSeries
+
+from ..basedatainterface import BaseDataInterface
+from ..baserecordingextractorinterface import (
+    BaseRecordingExtractorInterface
+)
+
+from spikeextractors.extractors.nwbextractors.nwbextractors import check_module
+
+
+# Helper functions for AxonaRecordingExtractorInterface
+def parse_generic_header(filename, params):
+    """
+    Given a binary file with phrases and line breaks, enters the
+    first word of a phrase as dictionary key and the following
+    string (without linebreaks) as value. Returns the dictionary.
+
+    INPUT
+    filename (str): .set file path and name.
+    params (list or set): parameter names to search for.
+
+    OUTPUT
+    header (dict): dictionary with keys being the parameters that
+                   were found & values being strings of the data.
+
+    EXAMPLE
+    parse_generic_header('myset_file.set', ['experimenter', 'trial_time'])
+    """
+    header = {}
+    params = set(params)
+    with open(filename, 'rb') as f:
+        for bin_line in f:
+            if b'data_start' in bin_line:
+                break
+            line = bin_line.decode('cp1252').replace('\r\n', '').\
+                replace('\r', '').strip()
+            parts = line.split(' ')
+            key = parts[0]
+            if key in params:
+                header[key] = ' '.join(parts[1:])
+
+    return header
+
+
+def read_iso_datetime(set_file):
+    """
+    Creates datetime object (y, m, d, h, m, s) from .set file header
+    and converts it to ISO 8601 format
+    """
+    with open(set_file, 'r', encoding='cp1252') as f:
+        for line in f:
+            if line.startswith('trial_date'):
+                date_string = re.findall(r'\d+\s\w+\s\d{4}$', line)[0]
```

Comment:
For things like this I always like to make 100% sure that this `date_string` format is something that is hard-coded by the acquisition system and not just something the user inputs?

Reply:
Good point. I noted it down for something to check with the Hussaini lab. More generally, it might then be safer to use something like this:

```
import dateutil.parser
yourdate = dateutil.parser.parse(datestring)
```

I tried it with a few different datestrings and it works nicely. Though even here, there can be ambiguous cases. 4.5.2021 can be read as 4th of May or 5th of April.